components/BugZapGame/BugZapLevel01.js

```diff
@@ -9,22 +9,165 @@ import {
 import reactMixin from 'react-mixin';
 import TimerMixin from 'react-timer-mixin';
 
-// import characters for animatedSprite to use
+// import characters for AnimatedSprite to use
 import frogCharacter from '../../sprites/frogLite/frogLiteCharacter';
 import bugCharacter from '../../sprites/bug/bugCharacter';
 import AnimatedSprite from "../AnimatedSprite/AnimatedSprite";
+import signCharacter from "../../sprites/sign/signCharacter"
 
 import styles from "./BugZapStyles";
 
 const SCREEN_WIDTH = require('Dimensions').get('window').width;
 const SCREEN_HEIGHT = require('Dimensions').get('window').height;
 
-const NUM_TRIALS = 3;
-// how long bug is catchable
-const BUG_IDLE_CATCH_DURATION = 1500;
+// const NUM_TRIALS = 3;
 
 class BugZapLevel01 extends React.Component {
-};
+  constructor (props) {
+    super(props);
+    this.state = {
+      frogKey: 0,
+      rotate: undefined,
+      frogPos: 250 * this.props.scale.width,
+      showOtherBug: false,
+      frogDirection: 'left',
+      level: 1,
+      tweenOptions: null,
+      startX: SCREEN_WIDTH/2 - (360 * this.props.scale.width),
+      showBug1: true,
+      frogAnimationIndex: [0],
+    };
+  }
+
+  componentDidMount () {
+    let direction = Math.floor(Math.random() * 2);
+    let frogDirection = 'left';
+
+    if (direction === 0){
+      frogDirection = 'right';
+      this.setState({
+        rotate: [{rotateY: '180deg'}],
+        frogPos: 500 * this.props.scale.width,
+        frogKey: Math.random(),
+        startX: SCREEN_WIDTH/2 + (210 * this.props.scale.width)
+      })
+    }
+
+    this.setState({
+      frogDirection: frogDirection,
+    });
+
+    this.setState({
+      tweenOptions: {
+        tweenType: "curve-fall",
+        startXY: [this.state.startX, 95 * this.props.scale.height], // start on their tags
+        endXY: [500 * this.props.scale.width, 460 * this.props.scale.height], // end at character
+        duration: 1000 * this.props.scale.width,
+        loop: false,
+      }
+    });
+  }
+
+  onTweenFinish (spriteKey) {
+    let index = [3];
+    if (this.state.frogDirection == "right") {
+      index = [1]
+    }
+    this.setState({
+      showBug1: false,
+      frogAnimationIndex: index,
+    });
+
+    this.goToNextTrial();
+  }
+
+  goToNextTrial () {
+    setTimeout (() => {
+      this.props.navigator.replace({
+        id: 'NextTrial',
+        getId: this.getCurrId, 
+      });
+    }, 500);
+  }
+
+  getCurrId () {
+    return 'BugZapLevel01';
+  }
+  
+
+  // onAnimationFinish (animationKey) {
+  //   console.warn('here');
+  //   this.setState({
+  //     frogAnimationIndex: [0],
+  //   })
+  // }
+
+  render () {
+    return (
+      <Image
+        source={require('../../backgrounds/Game_1_Background_1280.png')}
+        style={styles.backgroundImage} >
+
+      <AnimatedSprite
+        key={this.state.frogKey}
```

Comment:
Don't think you need to use key since not having multiple frogs and frog is not changed out.